daprdocs/content/en/developing-applications/building-blocks/state-management/howto-state-query-api.md

```diff
@@ -0,0 +1,140 @@
+---
+type: docs
+title: "How-To: Query the state"
+linkTitle: "How-To: Query the state"
+weight: 250
+description: "API for querying state stores"
+---
+
+{{% alert title="alpha" color="warning" %}}
+The state query API is in **alpha** stage.
+{{% /alert %}}
+
+## Introduction
+
+The state query API provides a way of querying the key/value data stored in state store components. This query API is not a replacement for a complete query language, and is focused on retrieving, filtering and sorting key/value data that you have saved through the state management APIs.
+
+This query API does not support querying of actor state stored in a state store. For that you need to use the query API for the specific database.
+See [querying actor state]({{< ref "state-management-overview.md#querying-actor-state" >}}).
+
+You can find additional information in the [resources]({{< ref "#resources" >}}) section.
+
+## Querying the state
+
+You submit query requests via HTTP POST/PUT or gRPC.
+The body of the request is the JSON map with 3 entries: `filter`, `sort`, and `pagination`.
+
+The `filter` section specifies the query conditions in the form of a tree of key/value operations, where the key is the operator and the value is the operands.
```

Comment:
Is filter a required section in the query?

Reply:
not required. If omitted, all entries will be returned.